crates/bevy_render/src/view/visibility/render_groups.rs

```diff
@@ -0,0 +1,678 @@
+    fn from(layer: RenderLayer) -> Self {
+        Self {
+            layers: RenderLayers::from(layer),
+            camera: None,
+        }
+    }
+}
+
+impl From<RenderLayers> for RenderGroups {
+    /// Makes a new `RenderGroups` from a [`RenderLayers`].
+    fn from(layers: RenderLayers) -> Self {
+        Self {
+            layers,
+            camera: None,
+        }
+    }
+}
+
+impl Default for RenderGroups {
+    /// Equivalent to `Self::from(DEFAULT_RENDER_LAYER)`.
+    fn default() -> Self {
+        Self::from(DEFAULT_RENDER_LAYER)
+    }
+}
+
+/// Stores a [`RenderGroups`] reference or value.
+///
+/// Useful for unwrapping an optional reference with fallback to a default value.
+pub enum RenderGroupsRef<'a> {
+    None,
+    Ref(&'a RenderGroups),
+    Val(RenderGroups),
+}
+
+impl<'a> RenderGroupsRef<'a> {
+    /// Passes [`RenderLists`] to `other` if `self` is on the heap.
+    ///
+    /// The ref will be in an invalid state after this is called.
+    pub(crate) fn reclaim(&mut self, other: &mut RenderGroups) {
+        match self {
+            Self::Val(groups) => {
+                if !groups.is_allocated() || other.is_allocated() {
+                    return;
+                }
+                std::mem::swap(groups, other);
+            }
+            _ => (),
+        }
+    }
+}
+
+impl<'a> Deref for RenderGroupsRef<'a> {
+    type Target = RenderGroups;
+
+    fn deref(&self) -> &Self::Target {
+        self.get()
+    }
+}
+
+impl<'a> RenderGroupsRef<'a> {
+    /// Gets a reference to the internal [`RenderGroups`].
+    ///
+    /// Panices if in state [`Self::None`].
+    pub fn get(&self) -> &RenderGroups {
+        match self {
+            Self::None => {
+                panic!("RenderGroupsRef cannot be dereferenced when empty");
+            }
+            Self::Ref(groups) => groups,
+            Self::Val(groups) => &groups,
+        }
+    }
+}
+
+/// Component on camera entities that controls which [`RenderLayers`] are visible to
+/// the camera.
+///
+/// A camera will see any entity that satisfies either of these conditions:
+/// - The entity is in a [`RenderLayer`] visible to the camera.
+/// - The entity has a [`RenderGroups`] component with camera affiliation equal to the camera.
+///
+/// Cameras use entities' [`InheritedRenderGroups`] to determine visibility, with a fall-back to the
+/// entity's [`RenderGroups`]. If an entity does not have [`InheritedRenderGroups`]
+/// or [`RenderGroups`] components, then the camera will only see it if the camera can
+/// view the [`DEFAULT_RENDER_LAYER`] layer.
+///
+/// A camera without the `CameraView` component will see the [`DEFAULT_RENDER_LAYER`]
+/// layer, in addition to any affiliated entities.
+///
+/// A default `CameraView` will include the [`DEFAULT_RENDER_LAYER`].
+#[derive(Component, Debug, Clone, PartialEq, Reflect)]
+#[reflect(Component, Default, PartialEq)]
+pub struct CameraView {
```

Comment:
Nit: I can't decide if "view" is too overloaded here. I totally understand the description of the component as "things I can see", but might get confused in it's relation to something like `ExtractedView` in render world.